code/game/objects/effects/glowshroom.dm

```diff
@@ -134,55 +131,45 @@
-			var/turf/newLoc = null
-
-			//Try three random locations to spawn before giving up tradeoff
-			//between running view(1, earth) on every single collected possibleLoc
-			//and failing to spread if we get 3 bad picks, which should only be a problem
-			//if there's a lot of glow shroom clustered about
-			for(var/Potato in 1 to 3)
-				var/turf/possibleLoc = pick(possibleLocs)
-				if(spreadsIntoAdjacent || !locate(/obj/structure/glowshroom) in view(1,possibleLoc))
-					newLoc = possibleLoc
-					break
-
-			//We failed to find any location, skip trying to yield
-			if(newLoc == null)
+	var/chance_generation = 100 * (NUM_E ** -(GLOWSHROOM_SPREAD_DIMINISH_FACTOR / myseed.potency * generation)) //https://www.desmos.com/calculator/a0oresnrwk
+
+	for(var/i in 1 to max(1, FLOOR(myseed.yield / 2, 1)))
+
+
+		if(!prob(chance_generation))
+			return
+		var/spreadsIntoAdjacent = prob(spreadIntoAdjacentChance)
+		var/turf/newLoc = null
+
+		//Try three random locations to spawn before giving up tradeoff
+		//between running view(1, earth) on every single collected possibleLoc
+		//and failing to spread if we get 3 bad picks, which should only be a problem
+		//if there's a lot of glow shroom clustered about
+		for(var/Potato in 1 to 3)
+			var/turf/possibleLoc = pick(possibleLocs)
+			if(spreadsIntoAdjacent || !locate(/obj/structure/glowshroom) in view(1,possibleLoc))
+				newLoc = possibleLoc
 				break
 
-			var/shroomCount = 0 //hacky
-			var/placeCount = 1
-			for(var/obj/structure/glowshroom/shroom in newLoc)
-				shroomCount++
-			for(var/wallDir in GLOB.cardinals)
-				var/turf/isWall = get_step(newLoc,wallDir)
-				if(isWall.density)
-					placeCount++
-			if(shroomCount >= placeCount)
-				continue
-
-			Decay(TRUE, GLOWCAP_ENDURANCE_SPREAD_COST) // Decay before spawning new mushrooms to reduce their endurance
-			if(QDELETED(src)) //Decay can end us
-				return
-			var/obj/structure/glowshroom/child = new type(newLoc, myseed, TRUE, TRUE)
-			child.generation = generation + 1
-			shrooms_planted++
-
-	if(!shrooms_planted)
-		max_failed_spreads--
-
-	//if we didn't get all possible shrooms planted or we haven't failed to spread at least 5 times then try to spread again later
-	if( (shrooms_planted <= myseed.yield) && (max_failed_spreads >= 0)  )
-		myseed.adjust_yield(-shrooms_planted)
-		//Lets make this a unique hash
-		addtimer(CALLBACK(src, .proc/Spread), delay_spread, TIMER_UNIQUE|TIMER_NO_HASH_WAIT)
+		//We failed to find any location, skip trying to yield
+		if(newLoc == null)
+			break
+
+		var/shroomCount = 0 //hacky
+		var/placeCount = 1
+		for(var/obj/structure/glowshroom/shroom in newLoc)
+			shroomCount++
+		for(var/wallDir in GLOB.cardinals)
+			var/turf/isWall = get_step(newLoc,wallDir)
+			if(isWall.density)
+				placeCount++
+		if(shroomCount >= placeCount)
+			continue
+
+		if(QDELETED(src)) //Decay can end us
+			return
+		var/obj/structure/glowshroom/child = new type(newLoc, myseed, TRUE, TRUE)
```

Comment:
Named arguments.

Reply:
Is there a contribution guideline for this? I've only ever seen this being brought up when you're using null inbetween args. I'm not saying its a bad suggestion, but it'd be nice if it could be formalized in the contribution guidelines if its not already.